src/main/java/biz/paluch/logging/gelf/log4j2/GelfLogAppender.java

```diff
@@ -284,6 +285,12 @@ public static GelfLogAppender createAppender(@PluginConfiguration final Configur
             ConfigurationSupport.setAdditionalFieldTypes(additionalFieldTypes, mdcGelfMessageAssembler);
         }
 
+        if (dynamicFieldTypeArray != null) {
+            for (GelfDynamicMdcFieldType gelfDynamicMdcFieldType : dynamicFieldTypeArray) {
+                mdcGelfMessageAssembler.setDynamicMdcFieldType(gelfDynamicMdcFieldType.getRegex(), gelfDynamicMdcFieldType.getType());
```

Comment:
It would make sense to add an overloaded `setDynamicMdcFieldType` method accepting also `Pattern` in addition to `String regex`.